templates/CRM/Contribute/Form/Contribution/Main.tpl

```diff
@@ -454,6 +454,17 @@
 
     $('#pricevalue').each(toggleBillingBlockIfFree).on('change', toggleBillingBlockIfFree);
   });
+
+  CRM.$(function($) {
+    // Update pledge contribution amount when pledge checkboxes change
+    cj("input[name^='pledge_amount']").on('change', function() {
+      var total = 0;
+      cj("input[name^='pledge_amount']:checked").each(function() {
+        total += Number(cj(this).attr('amount'));
+      });
+      cj("input[name^='price_']").val(total.toFixed(2));
```

Comment:
Hi Coleman,

 

cj is used throughout Main.tpl. Should I change all occurrences?

 

Marty

 

From: colemanw [mailto:notifications@github.com] 
Sent: Friday, October 30, 2015 3:11 PM
To: civicrm/civicrm-core
Cc: Marty Wright
Subject: Re: [civicrm-core] CRM-17469 - Online pledge payment fix (#7095)

 

In templates/CRM/Contribute/Form/Contribution/Main.tpl <https://github.com/civicrm/civicrm-core/pull/7095#discussion_r43552779> :

> @@ -454,6 +454,17 @@
>  
>      $('#pricevalue').each(toggleBillingBlockIfFree).on('change', toggleBillingBlockIfFree);
>    });
> +
> +  CRM.$(function($) {
> +    // Update pledge contribution amount when pledge checkboxes change
> +    cj("input[name^='pledge_amount']").on('change', function() {
> +      var total = 0;
> +      cj("input[name^='pledge_amount']:checked").each(function() {
> +        total += Number(cj(this).attr('amount'));
> +      });
> +      cj("input[name^='price_']").val(total.toFixed(2));

please change cj to $
In this context they both refer to jQuery but $ is more local and cj is deprecated.

—
Reply to this email directly or view it on GitHub <https://github.com/civicrm/civicrm-core/pull/7095/files#r43552779> .  <https://github.com/notifications/beacon/AOvUTMYjvAwYzAj16I6Gru9U2cJ0uOkoks5pA9RzgaJpZM4GZSC0.gif>